pandas/core/frame.py

```diff
@@ -1392,7 +1392,13 @@ def to_records(self, index=True, convert_datetime64=None):
             else:
                 if isinstance(self.index, MultiIndex):
                     # array of tuples to numpy cols. copy copy copy
-                    ix_vals = lmap(np.array, zip(*self.index.values))
+                    tuples = self.index.values
```

Comment:
so rather than doing this here, call ``.tolist()`` on this and fix the underlying code, like this:

```
In [30]: pd.MultiIndex.from_product([list('abc'), range(2)]).tolist()
Out[30]: [('a', 0), ('a', 1), ('b', 0), ('b', 1), ('c', 0), ('c', 1)]

In [31]: pd.MultiIndex.from_product([list('abc'), range(2)])[0:0].tolist()
Out[31]: []
```
so [31] should be a nested list arrays of the correct length/dtype

this might not work, but let's try

Reply:
yes you can override ``tolist()`` (its defined in pandas/core/base) in MultiIndex